### PR TITLE
Tanner ropp/create UI for server config

### PR DESCRIPTION
**Pull request by tanner-ropp**
# What does this PR do?
This PR adds the form for entering and editing server config fields in the problem editor.  There is validation for the three fields following the validation present in the back end.

Demo

#24

### CODE REVIEW

File: src/pages/ProblemEditor/ProblemEditorContainer/ProblemEditorContainer.tsx [hunk 1]
```diff
@@ -34,7 +37,7 @@ export const ProblemEditorContainer = () => {
   const formRef = useRef<FormikValues>();
 
   return (
-    <Box display="flex" flexDirection="column" flexGrow={1}>
+    <Box display="flex" flexDirection="column" flexGrow={1} textAlign="left">
```
Comment (marcgabe15) [inline src/pages/ProblemEditor/ProblemEditorContainer/ProblemEditorContainer.tsx:40]: does this work with RTL?

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "@mui/material": "^5.0.1",
     "@mui/styles": "^5.0.1",
     "@reduxjs/toolkit": "^1.5.1",
-    "@testing-library/jest-dom": "^4.2.4",
+    "@testing-library/jest-dom": "^5.14.1",
```
Comment (marcgabe15) [inline package.json:14]: What's new from version 4 to 5 that we have to upgrade?

**Review by marcgabe15 — APPROVED** [review]
Looks good! Just want clarification on the comments

File: src/pages/ProblemEditor/ProblemEditorContainer/ProblemEditorContainer.tsx (hunk 1 above)
Comment (tanner-ropp) [reply src/pages/ProblemEditor/ProblemEditorContainer/ProblemEditorContainer.tsx:40]: I'm not sure specifically what you are asking about

**Pull request merged**